docs/tutorials/create-an-ethereum-processing-squid.md

```diff
@@ -0,0 +1,455 @@
+
+The first thing to do, although it might sound trivial to GitHub experts, is to fork the repository into your own GitHub account, by visiting the [repository page](https://github.com/subsquid/squid-ethereum-template) and clicking the "Use this template" button:
+
+![How to fork a repository on GitHub](</img/.gitbook/assets/Screenshot-2022-02-02-111440.png>)
+
+Next, clone the created repository (be careful of changing `<account>` with your own account)
+
+```bash
+git clone git@github.com:<account>/squid-ethereum-template.git
+```
+
+For reference on the complete work, you can find the entire project [here](https://github.com/subsquid/squid-ethereum-example).
+
+## Define Entity Schema
+
+The next thing to do, in order to customize the project for our own purpose, is to make changes to the schema and define the Entities we want to keep track of.
+
+Luckily, the EVM template already contains a schema that defines the exact entities we need for the purpose of this guide. For this reason, changes are necessary, but it's still useful to explain what is going on.
+
+To index ERC-721 token transfers, we will need to track:
+
+* Token transfers
+* Ownership of tokens
+* Contracts and their minted tokens
+
+And the `schema.graphql` file defines them like shis:
+
+```graphql
+type Token @entity {
+  id: ID!
+  owner: Owner
+  uri: String
+  transfers: [Transfer!]! @derivedFrom(field: "token")
+  contract: Contract
+}
+ 
+type Owner @entity {
+  id: ID!
+  ownedTokens: [Token!] @derivedFrom(field: "owner")
+  balance: BigInt! @index
+}
+ 
+type Contract @entity {
+  id: ID!
+  name: String! @index
+  symbol: String! @index
+  # contract URI updated once e.g. a day
+  contractURI: String
+  address: String
+  # timestamp when the contract URI was updated last
+  contractURIUpdated: BigInt @index
+  totalSupply: BigInt!
+  mintedTokens: [Token!]! @derivedFrom(field: "contract")
+}
+ 
+type Transfer @entity {
+  id: ID!
+  token: Token!
+  from: Owner
+  to: Owner
+  timestamp: BigInt! @index
+  block: Int! @index
+  transactionHash: String! @index
+}
+```
+
+It's worth noting a couple of things in this [schema definition](https://docs.subsquid.io/reference/openreader-schema):
+
+* **`@entity`** - signals that this type will be translated into an ORM model that is going to be persisted in the database
+* **`@derivedFrom`** - signals the field will not be persisted on the database, it will rather be derived
+* **type references** (i.e. `from: Owner`) - establishes a relation between two entities
+* **`@index`** - signals thatt his field should be indexed by the database. Very useful for incresing performance on fields queried often.
+
+The template already has automatically generated TypeScript classes for this schema definition. They can be found under `src/model/generated`.
+
+Whenever changes are made to the schema, new TypeScript entity classes have to be generated, and to do that you'll have to run the `codegen` tool:
+
+```bash
+make codegen
```

Comment:
Better to write full command

Reply:
I usually do, but @dzhelezov seems to prefer we suggest commands in Makefile. Personally, as I have said a few times, I don't like either, and think we should have command lines in the CLI tool `sqd`

Reply:
I don't have a strong opinion here. I updated the migration guide to explicit commands. Generally, make seems easier to use, but explicit commands give more insights into what's going on under the hood and also don't require WSL on Windows. I think it doesn't make much difference, to be honest.